### PR TITLE
Node polyfills

### PR DESCRIPTION
**Pull request by Rich-Harris**
This replaces `@sveltejs/kit/install-fetch` with `@sveltejs/kit/node/polyfills`, and adds `crypto` to the list of web APIs that are guaranteed to be present.

* [ ] ~~add `AbortController` (#4846)~~
* [ ] add web streams (#3419)
* [ ] add other interfaces listed on [Minimum Common Web Platform API](https://common-min-api.proposal.wintercg.org/)? if we wanted to get really sophisticated we'd detect which APIs were used, so that adapters could only bundle those
* [ ] swap out `node-fetch` for `undici`, which is more standards compliant (though I think we're blocked on https://github.com/nodejs/undici/issues/1262)
* [x] only add polyfills if they're not already there (since Node 18 provides a lot of stuff anyway. in future maybe we should have `@sveltejs/kit/node/polyfills/16` and `@sveltejs/kit/node/polyfills/18`)
* [x] document which APIs are guaranteed to be present

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by ghostdevv** [comment]
Will this mean that libraries that rely on node dependencies in a non-node environment such as CloudFlare Pages have a chance of working? i.e. stripe

**Comment by Rich-Harris** [comment]
No, it's the opposite — it's bringing the Node environment up to date with other more web-like ones, where `fetch` and `crypto` etc are already universally supported. If people need to use Node-specific dependencies in non-Node environments, they will need to use things like https://www.npmjs.com/package/rollup-plugin-node-builtins (though some of these have significant caveats).

Ideally, those libraries should be updated to use web APIs, instead of assuming they're running in Node.

**Comment by ghostdevv** [comment]
Ah that makes sense, will try the link you gave but haven't had luck with similar packages and kit in the past :pray:

**Comment by Rich-Harris** [comment]
Gah. We can't use native `fetch` in Node 18 because it's secretly just undici, and we can't use undici because of https://github.com/nodejs/undici/issues/1262. That's why those tests are failing.

The response returned from undici's `fetch` omits `set-cookie` headers:

```js
import { createServer } from 'http';
// import fetch from 'node-fetch';

const PORT = 3000;

createServer((req, res) => {
  res.writeHead(200, {
    'set-cookie': 'foo=bar'
  });
  res.end('ok');
}).listen(PORT, async () => {
  const res = await fetch(`http://localhost:${PORT}`);
  console.log(res.headers.get('set-cookie'));
});
```

This logs `null`; if you uncomment the `import fetch from 'node-fetch'` it logs `foo=bar`.

I'm _pretty_ sure this is specific to undici (hopefully we can open a pull request), but we should probably check that you can read `set-cookie` in Deno and Cloudflare Workers (and the runtimes that sit on top of them — Netlify and Vercel edge functions).

**Comment by benmccann** [comment]
Just FYI, Vite is likely switching to `node-native-fetch`: https://github.com/vitejs/vite/pull/8157

**Comment by Rich-Harris** [comment]
Added `crypto` to the [web standards](https://kit.svelte.dev/docs/web-standards) section of the docs, so marking this ready for review. The other stuff (upgrading to `undici`, etc) can wait until another time, no point waiting on that for everything else.

File: packages/kit/src/node/index.js [hunk 1]
```diff
@@ -56,6 +57,7 @@ export async function getRequest(base, req) {
 	if (req.httpVersionMajor === 2) {
 		// we need to strip out the HTTP/2 pseudo-headers because node-fetch's
 		// Request implementation doesn't like them
+		// TODO is this still true with undici?
```
Comment (benmccann) [inline packages/kit/src/node/index.js:60]: maybe it'd be clearer to say Node 18 for people who aren't familiar with Undici?
```suggestion
		// TODO is this still true with Node 18
```

**Pull request merged**